File: Backend/src/main/java/com/luckyseven/backend/sharedkernel/jwt/entity/RefreshToken.java
package com.luckyseven.backend.sharedkernel.jwt.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RefreshToken {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long Id;

  private Long userId;

  private String tokenValue;

}

File: Backend/src/main/kotlin/com/luckyseven/backend/sharedkernel/jwt/entity/RefreshToken.kt
package com.luckyseven.backend.sharedkernel.jwt.entity

import com.luckyseven.backend.sharedkernel.jwt.util.JwtTokenizer
import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(name = "refresh_token", indexes = [
   Index(name = "idx_refresh_token_value", columnList = "tokenValue"),
   Index(name = "idx_refresh_token_user_id", columnList = "userId")
])
class RefreshToken(
   @Id
   @GeneratedValue(strategy = GenerationType.IDENTITY)
   val id: Long? = null,
   val userId: Long,
   @Column(length = 2000)
   val tokenValue: String,
   val issuedAt: Instant = Instant.now(),
   val expiresAt: Instant = Instant.now().plusMillis(JwtTokenizer.REFRESH_TOKEN_EXPIRE)
)